pkg/util/memory/meminfo.go

```diff
@@ -163,6 +170,37 @@ func init() {
 	terror.MustNil(err)
 }
 
+// InitMemoryHook initializes the memory hook.
+// It is to solve the problem that tidb cannot read cgroup in the systemd.
+// so if we are not in the container, we compare the cgroup memory limit and the physical memory,
+// the cgroup memory limit is smaller, we use the cgroup memory hook.
+func InitMemoryHook() {
```

Comment:
Can I ask why we need to have an init hook and an InitMemoryHook at the same time? Can we only put it into the explicit API or in the init function? Then we don't need to update the value in the different stages and places.

Reply:
```InitMemoryHook``` need to print log. but we cannot write log in the ```init```. logging have been inited. so we must run ```initMemoryHook``` after the logging's init.

Reply:
So maybe we can move all of them into InitMemoryHook.